### DISCUSSION TITLE
How can I parse mxnet params file into plain text?

### DISCUSSION THREAD

**Discussion by apepkuss** [apache/incubator-mxnet#14650] [Q&A]
## Description
I'm trying to parse mxnet params file into plain text in Python without using mxnet or gluon APIs. I tried the following simple Python code, but the result is not plain string, but looks like this "... \xaa>\x0f\xed\x8e>\xaf!\x8f>g ... " How can I get plain string from the param file? Thanks a lot.

## Environment info (Required)
Python = 3.6

```python
param_file = 'resnet-50-0000.params'
with open(param_file, 'rb') as f:
    net_params = f.read()
```

**Comment by vrakesh** [comment]
@apepkuss Thanks for your question, what is the purpose of wanting to read the params file if I may ask? Requesting more specifics on the use case.

@mxnet-label-bot  add [Question]

**Comment by vrakesh** [comment]
@mxnet-label-bot  add [Pending requester Info]

**Comment by haojin2** [comment]
@apepkuss Sorry for the late reply, what are you trying to do exactly? Are you trying to export the params in a human-readable format?

**Comment by vrakesh** [comment]
@apepkuss Requesting an update on the use-case

**Comment by ViswanathaReddyGajjala** [comment]
```python
import mxnet as mx

params = mx.nd.load('resnet.params')
idx = 0
for k, param in params.items():
    idx += 1
    print(idx, k, param.asnumpy().shape)
```
You can use the above code to view the params!
